subsys/bluetooth/audio/shell/bap.c

```diff
@@ -361,10 +330,19 @@ static void lc3_audio_timer_timeout(struct k_work *work)
 		}
 		lc3_sdu_cnt++;
 		seq_num++;
-	}
+	} while (lc3_sdu_cnt <= prime_count);
 }
 
-static K_WORK_DELAYABLE_DEFINE(audio_send_work, lc3_audio_timer_timeout);
+static K_WORK_DELAYABLE_DEFINE(audio_send_work, lc3_audio_send_data);
+
+void sdu_sent_cb(struct bt_bap_stream *stream)
```

Comment:
Previously we were either sending and the `k_work` was always active, which is why the `k_work_cancel` in `cmd_stop_sine` worked. However now when using the `sent` callback, this won't work, and we may need a `bool` to determine when to stop sending. Have you tested the `stop` command?